data/transactions/logic/eval.go

```diff
@@ -2325,6 +2325,33 @@ func opBytesNeq(cx *EvalContext) error {
 	return opNot(cx)
 }
 
+func opBytesModExp(cx *EvalContext) error {
+	result := new(big.Int)
+	last := len(cx.Stack) - 1 // z
+	prev := last - 1          // y
+	pprev := last - 2         // x
+
+	if len(cx.Stack[last].Bytes) > maxStringSize || len(cx.Stack[prev].Bytes) > maxStringSize || len(cx.Stack[pprev].Bytes) > maxStringSize {
+		return errors.New("bmodexp attempted on large byte-array")
+	}
+
+	z := new(big.Int).SetBytes(cx.Stack[last].Bytes)
+	y := new(big.Int).SetBytes(cx.Stack[prev].Bytes)
+	x := new(big.Int).SetBytes(cx.Stack[pprev].Bytes)
+	if z.BitLen() == 0 {
+		return errors.New("modulo by zero")
+	}
+
+	result.Exp(x, y, z)
+	if result.Sign() < 0 {
```

Comment:
This is impossible, since we only allow positive inputs, right?

Reply:
I'd remove it. `opBytesBin` needs to be able to handle things like `b-` going negative. `opBytesSqrt`, for example, does no similar check since it's a one-off implementation of the opcode.